README.markdown

```diff
@@ -381,30 +649,52 @@ Prints deprecation warnings and logs a warning once for a given key:
+*Type*: String, String.
 
-Additionally you can set the environment variable `STDLIB_LOG_DEPRECATIONS` to decide whether or not to log deprecation warnings: if this environment variable is set to `true`, the functions log a warning, if it is set to `false`, no warnings are logged. If no value is set at all, Puppet 4 will emit warnings, while Puppet 3 will not. Using this setting is especially useful for automated tests to avoid flooding your logs before you are ready to migrate.
+**Settings that affect `deprecation`**
 
-*Type*: String, String.
+Other settings in Puppet affect the stdlib `deprecation` function:
+
+* [`disable_warnings`](https://docs.puppet.com/puppet/latest/reference/configuration.html#disablewarnings)
+* [`max_deprecations`](https://docs.puppet.com/puppet/latest/reference/configuration.html#maxdeprecations)
+* [`strict`](https://docs.puppet.com/puppet/latest/reference/configuration.html#strict):
+
+    * `error`: Fails immediately with the deprecation message
+    * `off`: Output emits no messages.
+    * `warning`: Logs all warnings. This is the default setting.
+
+* The environment variable `STDLIB_LOG_DEPRECATIONS`
+
+  Specifies whether or not to log deprecation warnings. This is especially useful for automated tests to avoid flooding your logs before you are ready to migrate.
+
+  This variable is Boolean, with the following effects:
+  
+  * `true`: Functions log a warning.
+  * `false`: No warnings are logged.
+  * No value set: Puppet 4 emits warnings, but Puppet 3 does not.
 
 #### `difference`
 
-Returns the difference between two arrays. The returned array is a copy of the original array, removing any items that also appear in the second array. For example, `difference(["a","b","c"],["b","c","d"])` returns ["a"]. *Type*: rvalue.
+Returns the difference between two arrays. The returned array is a copy of the original array, removing any items that also appear in the second array.
 
-#### `dig`
+For example:
 
-DEPRECATED: This function has been replaced in Puppet 4.5.0, use dig44() for backwards compatibility or use the new version.
+* `difference(["a","b","c"],["b","c","d"])` returns ["a"].
 
 *Type*: rvalue.
 
+#### `dig`
+
+> DEPRECATED: This function has been replaced with a built-in [`dig`](https://docs.puppet.com/puppet/latest/function.html#dig) function as of Puppet 4.5.0. Use [`dig44()`](#dig44) for backwards compatibility or use the new version.
+
 Retrieves a value within multiple layers of hashes and arrays via an array of keys containing a path. The function goes through the structure by each path component and tries to return the value at the end of the path.
```

Comment:
I'd rather not, because my perspective is that even though we are warning them, if it still works, they should be able to find out how to use it. Open to discussion, of course.